app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardViewModel.kt

```diff
@@ -85,50 +95,55 @@ class SingleCardViewModel : ViewModel() {
+            val lCard =
+                localDatabase?.magicCardDao()?.getCard(card.set.code, card.number.toString())
+            if (lCard != null) {
+                // card owned -> make it not owned from the collection
+                if (lCard.possession == CardPossession.OWNED) {
+                    manageCardsInDatabase(card, CardPossession.NONE)
+                    updateButtons(CardPossession.NONE)
+                } else {
+                    // card not owned -> make it owned in the collection
+                    manageCardsInDatabase(card, CardPossession.OWNED)
+                    updateButtons(CardPossession.OWNED)
+                }
             } else {
-                addCardToFirebase(FBCardPossession.OWNED)
+                // card not in the database -> add it as owned in the local database
+                manageCardsInDatabase(card, CardPossession.OWNED)
+                updateButtons(CardPossession.OWNED)
             }
-        }.exceptionally {
-            addCardToFirebase(FBCardPossession.OWNED)
-            null
+            // sync the local database with the firebase if possible
+            DatabaseSync.sync()
         }
     }
 
     /**
-     * Add the card to the collection of the user
+     * Manage the cards in the local database
+     * @param card the card to manage
      * @param p the possession of the card (owned or wanted)
      */
-    private fun addCardToFirebase(p: FBCardPossession) {
-        val fbCard = FBMagicCard(card, p)
-        userDB.addCardToCollection(fbCard)
-        val data = userDB.getCardFromCollection(fbCard)
-        data.thenAccept {
-            if (p == FBCardPossession.OWNED) {
-                _buttonAddText.value = false
-                _buttonWantedText.value = true
-            } else {
-                _buttonWantedText.value = false
-
-            }
-        }
+    private suspend fun manageCardsInDatabase(card: MagicCard, p: CardPossession) {
+        localDatabase?.magicCardDao()?.insertCard(DBMagicCard(card, p))
     }
 
     /**
-     * Remove the card from the collection of the user
+     * Updates the wanted button text
      */
-    private fun removeCardFromFirebase() {
-        val fbCard = FBMagicCard(card)
-        userDB.removeCardFromCollection(fbCard)
-        val data = userDB.getCardFromCollection(fbCard)
-        data.exceptionally {
-            _buttonAddText.value = true
-            _buttonWantedText.value = true
-            null
+    private fun updateButtons(p: CardPossession) {
+        when (p) {
+            CardPossession.WANTED -> {
+                _buttonAddText.value = true
+                _buttonWantedText.value = false
+            }
+            CardPossession.OWNED -> {
+                _buttonWantedText.value = true
+                _buttonAddText.value = false
+            }
+            else -> {
+                _buttonWantedText.value = true
+                _buttonAddText.value = true
+            }

```

Comment:
simplify to "the added button is `p != OWNED` and the wanted button is `p != WANTED`" ?